src/main/java/com/couchbase/lite/replicator/PusherInternal.java

```diff
@@ -831,10 +841,11 @@ private void setPaused(boolean paused) {
 
     private void waitIfPaused(){
         synchronized (pausedObj) {
-            while (paused) {
+            while (paused && isRunning()) {
```

Comment:
@zgramana suggested `Paused` state should be one of StateMachine state. We should discuss about it. Ticket for it. https://github.com/couchbase/couchbase-lite-java-core/issues/1045

I will notify pauseObj when replicator state changed